src/charts/c3/c3-chart.component.js

```diff
@@ -97,9 +97,9 @@
           chartData = ctrl.config;
           if (chartData) {
             chartData.bindto = '#' + $attrs.id;
-            // always re-generate donut pct chart because it's colors
-            // change based on data and thresholds
-            if (!chart || $attrs.id.indexOf('donutPctChart')) {
+            // only re-generate donut pct chart if it has a threshold object
+            // because it's colors will change based on data and thresholds
+            if (!chart || ($attrs.id.includes('donutPctChart') && chartData.thresholds)) {
```

Comment:
Love to use this bit of newer ECMA, but the IE support appears like it's not there... an alternative, you could change it to use ```test``` and accomplish something similar...
```javascript
if (!chart || ( /donutPctChart/.test($attrs.id) && chartData.thresholds)) ...
```